packages/cadl-python/src/emitter.ts

```diff
@@ -721,6 +753,44 @@ function emitGlobalParameters(program: Program, namespace: NamespaceType): Recor
   }
 }
 
+function emitCredentialParam(program: Program, namespace: NamespaceType): Record<string, any> {
+  const auth = getAuthentication(program, namespace);
+  if (auth) {
+    for (const option of auth.options) {
+      for (const scheme of option.schemes) {
+        const type: CredentialType = {
+          kind: "Credential",
+          scheme: scheme,
+        }
+        const credential_type = getType(program, type);
+        if (credential_type) {
+          return {
+            type: credential_type,
+            optional: false,
+            description: "Credential needed for the client to connect to Azure.",
+            clientName: "credential",
+            location: "other",
+            restApiName: "credential",
+            implementation: "Client",
+            skipUrlEncoding: true,
+            inOverload: false
+          };
+        }
+      }
+    }
+  }
+  return {}
```

Comment:
it seems that the Prettier can't work although it doesn't influence running. I think we can configure some settings in .vscode to automatically format the code like `cadl-azure`.